### PR TITLE
Select table number

### PR DESCRIPTION
**Pull request by bilasa**
Navigates to the customer landing page upon clicking submit. Still needs functionality to retrieve table numbers from DB and store selected table number.

### CODE REVIEW

**Review by EBruchet — APPROVED** [review]
**Changes:**

- Added `serviceRequest` and `billRequest` columns to the `DiningSession` table in the database.
- Selecting a table number from the Customer page now updates the `diningSessionStatus` attribute of the `DiningSession` possessing the selected table number.
- Added list of auto-generated `DiningSessions` to `DatabaseLoader.java` (10)
- Converted the table selection dropdown into its own component `CustomerDiningSessionSelect`, placed in `DiningSession.js`
- Shifted the table number selection view into its own component `TableNumberSelect` out of the `render` method in which it was initially found.
- List of tables to select from is no longer static, pulls from the `DiningSession` table database.

Merge pending final review from @Gabetn

**Review by Gabetn — CHANGES_REQUESTED** [review]
### Gitignore
*.DS_Store* is improperly being tracked. It's being generated by mac users and should be removed from this PR as well as added to the .gitignore.
_Note_: I will do this. 

### Creating new tables
Currently implemented via drop-down selection and auto-populated via the database loader.
Should we include functionality for creating a table within the system?  i.e. if we wanted to add a 11th table.
_*Note*_: User story 12 is a bit vague in this regard. 

### Selecting already assigned ID
These table numbers are unique, however currently we can assign an already assigned table.
* Option 1: Send error if selecting already assigned table
* Option 2: Only display in the drop-down list the un-assigned tables

**Implementation restriction:** Currently status refers to a dining-session status and gets reset when the bill is paid. However in the case where the bill is paid and the status is set to INACTIVE, how can we parse our list of assigned table#'s to avoid duplication. Since if we only parse by status we can potentially have multiple tablets with the same table# if they are all added when the diningsession for that table number is inactive.

I will think about it, but suggestions are welcome.

Aside from those I'll continue testing the functionality.

File: src/main/js/components/Customer.js
Comment (Gabetn) [inline src/main/js/components/Customer.js:77]: When we push the customerLanding page onto the history, how are the props propagated?

My specific issue is that looking long-term when the customer creates an order they will need to know their diningsession, but there can be multiple active.

As such I think the customerLanding should have an additional property **_tableNumber_**. So that way the customer when creating an order can access its dining session via `this.props.tableNumber` and then parse the `diningsessions` for the corresponding **_tableNumber_**. 

However with the current implementation I don't see how we can know which dining session corresponds to the active tablet on which the session is ran.

Comment (EBruchet) [reply src/main/js/components/Customer.js:77]: Sounds good with me

**Review by Gabetn — APPROVED** [review]

**Comment by Gabetn** [comment]
## UPDATE
### Gitignore
Has been modified and the history of .DS_Store has been purged. 

### Creating new tables
Low-priority issue. I will create an issue for it so we can keep track and decide on whether to implement it at the end. 

### Selecting Assigned ID
**SOLVED**: Added column `ta_status` to `DiningSession` table in db to keep track of which table#'s have been assigned _regardless_ of the status of the _diningsession_. 
This is accomplished via the filtering of diningsessions, see `filterDiningSessionList` in **App.js**. I made it "polymorphic" for use in the Staff view

### Pushing tableNumber to main customer view after manager selects
_See_  my comment in *Customer.js* for background
**TODO**: fix after merging in modified `Customer.js` from PR #13 
I will create an issue for this as a reminder

**Pull request merged**
